### PR TITLE
fix(hooks,ui): display payload from gerrit + fix hook edit

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: ui/src/app/store/workflow.state.ts [hunk 1]
```diff
@@ -202,9 +202,18 @@ export class WorkflowState {
                 wf.template_up_to_date = cloneDeep(oldWorkflow.template_up_to_date);
                 wf.as_code_events = cloneDeep(oldWorkflow.as_code_events);
 
+                // Generate hook ref for UI edition
+                if (wf && wf.workflow_data && wf.workflow_data.node && wf.workflow_data.node.hooks) {
+                    wf.workflow_data.node.hooks.forEach(h => {
+                        if (!h.ref) {
```
Comment (richardlt) [inline ui/src/app/store/workflow.state.ts]: h is a copy or ref ?

File: ui/src/app/store/workflow.state.ts [hunk 2]
```diff
@@ -225,11 +234,20 @@ export class WorkflowState {
                 wfUpdated.template_up_to_date = state.workflow.template_up_to_date;
                 wfUpdated.as_code_events = state.workflow.as_code_events;
 
+                // Generate hook ref for UI edition
+                if (wfUpdated.workflow_data && wfUpdated.workflow_data.node && wfUpdated.workflow_data.node.hooks) {
+                    wfUpdated.workflow_data.node.hooks.forEach(h => {
+                        if (!h.ref) {
```
Comment (richardlt) [inline ui/src/app/store/workflow.state.ts]: same

File: ui/src/app/store/workflow.state.ts [hunk 3]
```diff
@@ -849,6 +867,16 @@ export class WorkflowState {
                 let canEdit = wf.permissions.writable;
                 let editWorkflow: Workflow;
                 let editMode: boolean;
+
+                // Generate hook ref for UI edition
+                if (wf && wf.workflow_data && wf.workflow_data.node && wf.workflow_data.node.hooks) {
+                    wf.workflow_data.node.hooks.forEach(h => {
+                        if (!h.ref) {
```
Comment (richardlt) [inline ui/src/app/store/workflow.state.ts]: same

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12410.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: ui/src/app/store/workflow.state.ts (hunk 1 above)
Comment (sguiheux) [reply ui/src/app/store/workflow.state.ts]: forEach -> ref

File: ui/src/app/store/workflow.state.ts (hunk 2 above)
Comment (sguiheux) [reply ui/src/app/store/workflow.state.ts]: forEach -> ref

File: ui/src/app/store/workflow.state.ts (hunk 3 above)
Comment (sguiheux) [reply ui/src/app/store/workflow.state.ts]: forEach -> ref

**Comment by sguiheux** [comment]
@yesnault There is no documentation on gerrit integration. It will be done in another PR

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12410.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Pull request merged**